iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/helplets.inl

```diff
@@ -128,21 +129,25 @@ inline bool isValidPathToDirectory(const string<StringCapacity>& name) noexcept
         // with a valid file name
         if (separatorPosition)
         {
-            auto filenameToVerify = temp.substr(0, *separatorPosition);
-            bool isValidDirectory = isValidFileName(*filenameToVerify) || *filenameToVerify == currentDirectory
-                                    || *filenameToVerify == parentDirectory;
+            auto filenameToVerify{temp.substr(0, *separatorPosition)};
+            bool isValidDirectory{isValidFileName(*filenameToVerify) || *filenameToVerify == currentDirectory
+                                  || *filenameToVerify == parentDirectory};
             if (!isValidDirectory)
             {
                 return false;
             }
 
-            temp.substr(*separatorPosition + 1).and_then([&](auto& s) { temp = s; });
+            temp.substr(*separatorPosition + 1).and_then([&temp](auto& s) { temp = s; });
         }
         // we reached the last entry, if its a valid file name the path is valid
         else if (!separatorPosition)
         {
             return isValidPathEntry(temp, RelativePathComponents::ACCEPT);
         }
+        else
+        {
+            // do nothing
+        }
```

Comment:
Have you tried:
```suggestion
        else
        {
            return isValidPathEntry(temp, RelativePathComponents::ACCEPT);
        }
```
?

Reply:
@FerdinandSpitzschnueffler not yet, let me try now.

Reply:
@FerdinandSpitzschnueffler not yet, let me try.